lib/testutils/src/lib.rs

```diff
@@ -383,61 +383,136 @@ pub fn write_file(store: &Store, path: &RepoPath, contents: &str) -> FileId {
         .unwrap()
 }
 
-pub fn write_normal_file(
-    tree_builder: &mut TreeBuilder,
-    path: &RepoPath,
-    contents: &str,
-) -> FileId {
-    let id = write_file(tree_builder.store(), path, contents);
-    tree_builder.set(
-        path.to_owned(),
-        TreeValue::File {
-            id: id.clone(),
+pub struct TestTreeBuilder {
+    store: Arc<Store>,
+    tree_builder: TreeBuilder,
+}
+
+impl TestTreeBuilder {
+    pub fn new(store: Arc<Store>) -> Self {
+        let tree_builder = store.tree_builder(store.empty_tree_id().clone());
+        Self {
+            store,
+            tree_builder,
+        }
+    }
+
+    pub fn entry(&mut self, path: &RepoPath) -> TestTreeEntryBuilder<'_> {
+        TestTreeEntryBuilder {
+            tree_builder: &mut self.tree_builder,
+            path: path.to_owned(),
+        }
+    }
```

Comment:
nit: I think we can simply add `binary_file(path, content)`, `symlink(path, target)`, etc. here.

Sorry, I just said `.entry(path)` without considering how the API would look like.